src/plugins/crypto/crypto.h

```diff
@@ -13,22 +13,49 @@
 #include <kdbplugin.h>
 #include <stdio.h>
 
-#include <crypto_internal.h>
-
-// methods for kdb
-int elektraCryptoOpen(Plugin *handle, Key *errorKey);
-int elektraCryptoClose(Plugin *handle, Key *errorKey);
-int elektraCryptoGet(Plugin *handle, KeySet *ks, Key *parentKey);
-int elektraCryptoSet(Plugin *handle, KeySet *ks, Key *parentKey);
-int elektraCryptoError(Plugin *handle, KeySet *ks, Key *parentKey);
-
-// provider-independent crypto functions
-int elektraCryptoInit(Key *errorKey);
-void elektraCryptoTeardown();
-int elektraCryptoHandleCreate(elektraCryptoHandle **handle, KeySet *config, Key *errorKey);
-void elektraCryptoHandleDestroy(elektraCryptoHandle *handle);
-int elektraCryptoEncrypt(elektraCryptoHandle *handle, Key *k, Key *errorKey);
-int elektraCryptoDecrypt(elektraCryptoHandle *handle, Key *k, Key *errorKey);
+enum ElektraCryptoHeaderFlags
+{
+	ELEKTRA_CRYPTO_FLAG_NONE = 0,
+	ELEKTRA_CRYPTO_FLAG_STRING = 1,
+	ELEKTRA_CRYPTO_FLAG_NULL = 2
+};
+
+#define ELEKTRA_CRYPTO_META_ENCRYPT ("crypto/encrypt")
+#define ELEKTRA_CRYPTO_META_ENCRYPTED ("crypto/encrypted")
```

Comment:
Its important that you think through what could go wrong and what a user expect in these situations. Failing `kdbGet()` is quite drastic, because then an application cannot start. (Failing `kdbSet()` is less so problematic.) Most problems will be caused by key derivation, nevertheless it is good if you keep error handling in mind already.